cmd/quadlet/main.go

```diff
@@ -107,31 +107,33 @@ func Debugf(format string, a ...interface{}) {
 // For system generators these are in /usr/share/containers/systemd (for distro files)
 // and /etc/containers/systemd (for sysadmin files).
 // For user generators these can live in $XDG_RUNTIME_DIR/containers/systemd, /etc/containers/systemd/users, /etc/containers/systemd/users/$UID, and $XDG_CONFIG_HOME/containers/systemd
-func getUnitDirs(rootless bool) []string {
-	// Allow overriding source dir, this is mainly for the CI tests
-	if varExist, dirs := getDirsFromEnv(); varExist {
-		return dirs
-	}
+func getUnitDirs(rootless bool) map[string]struct{} {
+	dirs := make(map[string]struct{}, 0)
 
-	resolvedUnitDirAdminUser := resolveUnitDirAdminUser()
-	userLevelFilter := getUserLevelFilter(resolvedUnitDirAdminUser)
-
-	if rootless {
-		systemUserDirLevel := len(strings.Split(resolvedUnitDirAdminUser, string(os.PathSeparator)))
-		nonNumericFilter := getNonNumericFilter(resolvedUnitDirAdminUser, systemUserDirLevel)
-		return getRootlessDirs(nonNumericFilter, userLevelFilter)
+	// Allow overriding source dir, this is mainly for the CI tests
+	varExist, dirs := getDirsFromEnv(dirs)
+	if !varExist {
+		resolvedUnitDirAdminUser := resolveUnitDirAdminUser()
+		userLevelFilter := getUserLevelFilter(resolvedUnitDirAdminUser)
+
+		if rootless {
+			systemUserDirLevel := len(strings.Split(resolvedUnitDirAdminUser, string(os.PathSeparator)))
+			nonNumericFilter := getNonNumericFilter(resolvedUnitDirAdminUser, systemUserDirLevel)
+			dirs = getRootlessDirs(dirs, nonNumericFilter, userLevelFilter)
+		} else {
+			dirs = getRootDirs(dirs, userLevelFilter)
```

Comment:
same idea could be used here i think? early return.

Reply:
You're right. But, I have to say in the previous implementation the return inside the `rootless` statement kinda confused me. So, I rather have a single return when it's not really an early return but rather a different path